### PR TITLE
Attach documents to plain-text emails

### PR DESCRIPTION
**Pull request by servancho**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

Very useful feature for Plain Text emails - send documents as attachments with follow-up emails.

*Please update this template with something that matches your PR*

### CODE REVIEW

**Comment by trasher** [comment]
Please rebase your branch onto an up to date 9.2/bugfixes branch

**Comment by servancho** [comment]
Rebased.

**Comment by servancho** [comment]
CircleCI was fixed. Thank you for your software )

**Comment by servancho** [comment]
@orthagh code fragment was refactored.

File: inc/notificationeventmailing.class.php [hunk 1]
```diff
@@ -137,27 +137,38 @@ static public function send(array $data) {
...
-            ]);
             $inline_docs = [];
-            $doc = new Document();
             if (count($document_items)) {
                foreach ($document_items as $doc_i_data) {
-                  $doc->getFromDB($doc_i_data['documents_id']);
                   // Add embeded image if tag present in ticket content
                   if (preg_match_all('/'.Document::getImageTag($doc->fields['tag']).'/',
                                      $current->fields['body_html'], $matches, PREG_PATTERN_ORDER)) {
+
```
Comment (orthagh) [inline inc/notificationeventmailing.class.php]: This part should be moved above l167 (`if (preg_match_all('/'.Document::getImageTag($doc->fields['tag']).'/',`)

Comment (servancho) [reply inc/notificationeventmailing.class.php]: What exactly shuld be removed? You mean to put cycle in attachDocuments and use lambda for inner logic?

**Comment by servancho** [comment]
@orthagh please, review latest changes. It was tested in production.

File: inc/commondbtm.class.php [hunk 2]
```diff
@@ -491,7 +491,7 @@ function addToDB() {
                if (($this->getType() == 'ProfileRight') && ($values[$i] == '')) {
                   $values[$i] = 0;
                }
-               $query .= "'".$values[$i]."'";
+               $query .= "'".htmlspecialchars($values[$i])."'";
```
Comment (trasher) [inline inc/commondbtm.class.php]: This is not correct, and this breaks tests. Please undo.

**Comment by trasher** [comment]
Replaced with #6471

**Pull request closed without merging**